File: 83/Solution.java
class Solution {
    public class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) {
            return head;
        }
        ListNode cur = head;
        ListNode nxt = head.next;
        while (nxt != null) {
            while (nxt != null && cur.val == nxt.val) {
                nxt = nxt.next;
                cur.next = nxt;
            }
            if (nxt != null) {
                cur = nxt;
                nxt = nxt.next;
            }
        }
        return head;
    }
}

File: 83/solution.go
package _83

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	l, r := head, head
	for r != nil {
		if l.Val != r.Val {
			l.Next = r
			l = r
		}
		r = r.Next
	}
	l.Next = nil
	return head
}
